SpatialGDK/Source/SpatialGDK/Private/Schema/ChangeInterest.cpp

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Schema/ChangeInterest.h"
+
+DEFINE_LOG_CATEGORY(LogChangeInterest);
+
+namespace SpatialGDK
+{
+void ChangeInterestQuery::DebugOutput(const FString& DiffType) const
+{
+	// Minimal output
+	if (Components.Find(SpatialConstants::CLIENT_ENDPOINT_COMPONENT_ID) != INDEX_NONE)
+	{
+		UE_LOG(LogChangeInterest, Log, TEXT("Interest diff: client auth interest"));
+	}
+	else
+	{
+		UE_LOG(LogChangeInterest, Log, TEXT("Interest diff: client non auth interest"));
+	}
+
+	/*
+	// Verbose output
+	FString ComponentsString;
+	for (const auto& Component : Components)
+	{
+		ComponentsString += FString::Format(TEXT("{0} "), { Component });
+	}
+
+	UE_LOG(LogChangeInterest, Log, TEXT("Interest diff: components %s"), *ComponentsString);
+
+	FString ComponentSetsString;
+	for (const auto& ComponentSet : ComponentSets)
+	{
+		ComponentSetsString += FString::Format(TEXT("{0} "), { ComponentSet });
+	}
+
+	UE_LOG(LogChangeInterest, Log, TEXT("Interest diff: component sets %s"), *ComponentSetsString);
+	*/
```

Comment:
thoughts on putting this in / leaving out?

Reply:
shouldn't be committed to mainline - but seems useful while we're still debugging the implementation